core/iwasm/common/wasm_memory.c

```diff
@@ -994,7 +989,9 @@ wasm_enlarge_memory_with_idx(WASMModuleInstance *module, uint32 inc_page_count,
     if (memidx < module->memory_count)
         shared_memory_lock(module->memories[memidx]);
 #endif
-    ret = wasm_enlarge_memory_internal(module, inc_page_count, memidx);
+    ret =
+        wasm_enlarge_memory_internal((WASMModuleInstanceCommon *)module,
+                                     module->memories[memidx], inc_page_count);
```

Comment:
If it will be an export API, we also need to check whether `module->memories[memidx` is valid. Ignore me if this won't be called in export API :)

Reply:
I think after the fixes I made for your previous comments, there are no longer any cases that can call `wasm_enlarge_memory_internal()` without first checking that the memory instance arg is valid. Please let me know if you think I missed something though. However, the first block of code in `wasm_enlarge_memory_internal()` checks that memory is valid so I think it should be safe regardless.

```
    if (!memory) {
        ret = false;
        goto return_func;
    }
```